### PR TITLE
rename import_source to render_source

### PR DESCRIPTION
**Pull request by guswynn**
(broken out from https://github.com/MaterializeInc/materialize/pull/12003)

### Motivation

Docs

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

None

### CODE REVIEW

**Review by petrosagg — APPROVED** [review]

**Comment by petrosagg** [comment]
Are the item visibility changes required?

File: src/storage/src/render/sources.rs [hunk 1]
```diff
@@ -127,11 +129,13 @@ where
...
 }
 
-/// Constructs a `CollectionBundle` and tokens from source arguments.
+/// Constructs a data [`Collection`], an error [`Collection`],
+/// and tokens from a [`SourceInstanceDesc`].
 ///
 /// The first returned pair are the row and error collections, and the
 /// second is a token that will keep the source alive as long as it is held.
-pub(crate) fn import_source<G>(
+// TODO(guswynn): write docs inline for this function
```
Comment (benesch) [inline src/storage/src/render/sources.rs:137]: ❤️

**Comment by guswynn** [comment]
@petrosagg no, I can clean those up!

**Comment by guswynn** [comment]
actually, the `render_source` pub change is needed~

**Pull request merged**